### PR TITLE
fix pillow retry for new-style pillows

### PR DESCRIPTION
**Pull request by czue**
http://manage.dimagi.com/default.asp?221319

@snopoke :fish: 

the first 18 commits (heh) are cleanup and prework. the last one has the change.

### CODE REVIEW

File: corehq/ex-submodules/pillowtop/management/commands/ptop_reset_checkpoint.py [hunk 1]
```diff
@@ -42,7 +41,7 @@ def handle(self, *labels, **options):
             operation, and may take a long time, and cause extraneous updates to the requisite
             consumers of the _changes feeds  Are you sure you want to do this?
 
-Type 'yes' to continue, or 'no' to cancel: """  % pillow_class_name)
+Type 'yes' to continue, or 'no' to cancel: """  % pillow_name)
```
Comment (dimagimon) [inline corehq/ex-submodules/pillowtop/management/commands/ptop_reset_checkpoint.py]: E221 multiple spaces before operator

**Comment by benrudolph** [comment]
```
======================================================================
ERROR: testCaseAPIResultJSON (corehq.apps.cloudcare.tests.test_api.CaseAPITest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/corehq/apps/cloudcare/tests/test_api.py", line 31, in setUp
    create_domain(self.domain)
  File "/mnt/corehq/apps/domain/shortcuts.py", line 9, in create_domain
    secure_submissions=False)
  File "/mnt/corehq/apps/domain/models.py", line 553, in get_or_create_with_name
    new_domain.save(**get_safe_write_kwargs())
  File "/mnt/corehq/apps/domain/models.py", line 611, in save
    super(Domain, self).save(**params)
  File "/mnt/corehq/apps/cachehq/mixins.py", line 17, in save
    self.clear_caches()
  File "/mnt/corehq/apps/domain/models.py", line 1060, in clear_caches
    super(Domain, self).clear_caches()
  File "/mnt/corehq/apps/cachehq/mixins.py", line 58, in clear_caches
    super(QuickCachedDocumentMixin, self).clear_caches()
  File "/mnt/corehq/apps/cachehq/mixins.py", line 11, in clear_caches
    invalidate_document(self, couch_db=self.get_db())
  File "/mnt/corehq/apps/cachehq/invalidate.py", line 16, in invalidate_document
    get_cache_pillow(couch_db).change_trigger({
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 11, in _inner
    return m(*args, **kwargs)
  File "submodules/dimagi-utils-src/dimagi/utils/decorators/memoized.py", line 128, in __call__
    value = self.func( *args, **kwargs)
  File "/mnt/corehq/apps/cachehq/invalidate.py", line 7, in get_cache_pillow
    return cacheinvalidate.CacheInvalidatePillow(couch_db=couch_db)
TypeError: __init__() takes at least 3 arguments (2 given)
```

**Comment by benrudolph** [comment]
this is great! thanks for cleanup and fix

**Comment by czue** [comment]
thx, will fix

File: corehq/ex-submodules/pillowtop/utils.py [hunk 2]
```diff
@@ -27,7 +27,7 @@ def _get_pillow_class(full_class_str):
     except (AttributeError, ImportError):
         if settings.DEBUG:
             raise
-        raise ValueError("Could not find pillowtop class '%s'" % full_class_str)
+        raise ValueError("Could not find class or function: '%s'" % full_class_str)
```
Comment (snopoke) [inline corehq/ex-submodules/pillowtop/utils.py]: This is basically identical to `dimagi.utils.modules.to_function`

**Comment by snopoke** [comment]
:+1:

File: corehq/ex-submodules/pillowtop/utils.py (hunk 2 above)
Comment (czue) [inline corehq/ex-submodules/pillowtop/utils.py]: good point. https://github.com/dimagi/commcare-hq/pull/10816/commits/1ee66be72b69b6fb75f08bc5990b8668b6579fa4 https://github.com/dimagi/dimagi-utils/pull/354

Comment (snopoke) [inline corehq/ex-submodules/pillowtop/utils.py]: :+1:

**Comment by dannyroberts** [comment]
```
======================================================================
ERROR: test_get_active_case_counts (corehq.apps.reports.tests.test_esaccessors.TestCaseESAccessorsSQL)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/corehq/apps/reports/tests/test_esaccessors.py", line 768, in setUp
    super(TestCaseESAccessors, self).setUp()
  File "/mnt/corehq/apps/reports/tests/test_esaccessors.py", line 61, in setUp
    self.pillow = self.get_pillow()
  File "/mnt/corehq/apps/reports/tests/test_esaccessors.py", line 940, in get_pillow
    return get_sql_case_to_elasticsearch_pillow()
TypeError: get_sql_case_to_elasticsearch_pillow() takes exactly 1 argument (0 given)
```

**Comment by czue** [comment]
thx. feels good to trust my tests

**Comment by czue** [comment]
:white_check_mark: finally

**Pull request merged**